### PR TITLE
Initial typescript setup

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

- The start of https://github.com/Flagsmith/flagsmith/issues/1707, adds typescript to the webpack, babel and converts button to a tsx component that passes compilation.
- Adds a base config for webpack for others to extend from
- Cleans up props of button by removing unnecessary buttonText
- Converts button components from class to functional components

## How did you test this code?
Ran and the project in production/development mode. Ran the newly added ``npm run typecheck``

### CODE REVIEW

File: .github/actions/e2e-tests/action.yml [hunk 1]
```diff
@@ -43,6 +43,9 @@ runs:
           ENABLE_INFLUXDB_FEATURES: ${{ inputs.enable_influxdb_features }}
 
         run: |
+          wget -q https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_96.0.4664.110-1_amd64.deb
```
Comment (matthewelwell) [inline .github/actions/e2e-tests/action.yml]: I'm sure this was only recently removed by @dabeeeenster, how come it's being re-added?

File: frontend/common/services/useSegment.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+export const {
+  useGetSegmentsQuery,
+  useDeleteSegmentMutation,
+  useUpdateSegmentMutation,
+  useCreateSegmentMutation,
+  // END OF EXPORTS
+} = segmentService
+
+// const { data, isLoading } = useGetSegmentsQuery({ id: 2 }, {}) get hook
```
Comment (matthewelwell) [inline frontend/common/services/useSegment.ts]: How come we've got a bunch of commented out code here?

File: frontend/web/components/modals/CreateSegment.js [hunk 3]
```diff
@@ -435,10 +435,12 @@ const LoadingCreateSegment  = (props) => {
     const [segmentData, setSegmentData] = useState(null);
 
     useEffect(()=>{
-        _data.get(`${Project.api}projects/${props.projectId}/segments/${props.segment}`).then((segment)=> {
-            setSegmentData(segment);
-            setLoading(false)
-        })
+        if(props.segment) {
```
Comment (matthewelwell) [inline frontend/web/components/modals/CreateSegment.js]: Didn't we add this recently? Do you need to merge main back into this branch perhaps?

File: frontend/common/services/useSegment.ts (hunk 2 above)
Comment (kyle-ssg) [reply frontend/common/services/useSegment.ts]: Usage example

File: frontend/web/components/modals/CreateSegment.js (hunk 3 above)
Comment (kyle-ssg) [reply frontend/web/components/modals/CreateSegment.js]: Ah, yep I think so

File: frontend/common/services/useSegment.ts (hunk 2 above)
Comment (matthewelwell) [reply frontend/common/services/useSegment.ts]: Ah, ok - the other comments with usage examples seem to have a 'usage example' prefix. e.g. https://github.com/Flagsmith/flagsmith/pull/1708/files#diff-1f3cc3a02c13bfbc2f59ddf91d03fd4a29cfc247830f1c0082bbd7ee2176215dR71

Comment (kyle-ssg) [reply frontend/common/services/useSegment.ts]: I'll adjust, the CLI generates this so I'll adjust that too - I do prefer that format yeah

File: .github/actions/e2e-tests/action.yml (hunk 1 above)
Comment (kyle-ssg) [reply .github/actions/e2e-tests/action.yml]: I'll revert this, I bet the concurrency will sort the freezes

File: frontend/common/services/useSegment.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                query: (query: Req['deleteSegment']) => ({
+                    url: `projects/${query.projectId}/segments/${query.id}/`,
+                    method: 'DELETE',
+                    body: query,
+                }),
+                invalidatesTags: (q, e, arg) => [{ type: 'Segment', id: `LIST${arg.projectId}` }],
+            }),
+            updateSegment: builder.mutation<Res['segment'], Req['updateSegment']>({
+                query: (query: Req['updateSegment']) => ({
+                    url: `projects/${query.projectId}/segments/${query.id}`,
```
Comment (matthewelwell) [inline frontend/common/services/useSegment.ts]: It would be a good idea to add the trailing slash here too.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**